### PR TITLE
fix some components after breaking changes upstream

### PR DESCRIPTION
**Pull request by jchavarri**
This PR updates a few things:

- Adds `github_username` to project creation prompt (as it seemed to be used later on)
- Fixes usages of `React.*` modules after breaking changes in jsoo-react
- Updated contributing docs to reflect opam install
- Remove duplicated dev-dependencies entries and unused `bs-*` deps.

@tmattio let me know if you prefer to take some of these things aside in a different PR :)

### CODE REVIEW

File: template/Makefile [hunk 1]
```diff
@@ -45,6 +45,11 @@ dev: ## Install development dependencies
 	npm install
 	opam install --deps-only --with-test --with-doc -y .
 
+.PHONY: create_switch
+create_switch:
+  # Create a local opam switch
+	opam switch create . 4.10.0 --deps-only --with-test
```
Comment (tmattio) [inline template/Makefile]: I think you'll want to use `--no-install` when creating the switch, because some of the dependencies need to be pinned first.

**Comment by tmattio** [comment]
Thanks! It looks great, I just left a comment about the switch creation 🙂

File: template/Makefile [hunk 2]
```diff
@@ -45,6 +45,11 @@ dev: ## Install development dependencies
 	npm install
 	opam install --deps-only --with-test --with-doc -y .
 
+.PHONY: create_switch
+create_switch:
+  # Create a local opam switch
```
Comment (tmattio) [inline template/Makefile]: I think the comment needs to be in the same line as the name of the command for the regex to catch it and display the doc in `make help`:

```
.PHONY: create_switch
create_switch: ## Create a local opam switch
  ...
```

**Comment by tmattio** [comment]
> Adds github_username to project creation prompt (as it seemed to be used later on)

Make me think: Spin should probably validate that all the variables used in the template definition are defined. Will open an issue.

**Comment by jchavarri** [comment]
@tmattio great points, fixed them in last commit.

One question: do you have any specific workflow to test Spin templates while iterating over them? I find myself very "in the dark" because I apply the changes in a project created with the template, and then apply them "backwards" to the template manually, but it feels too easy to forget something in that process. I wonder if there's a better way 🤔

**Comment by tmattio** [comment]
@jchavarri I'm doing pretty much the same. When I first develop a template, I start with the non-templatized project and port it to a spin template once it's ready, but when the template already exists, I am working on the generated project, check the diff, and apply it to the template manually.

It's definitely a pain point for me as well, so I'm open to suggestions if you have an idea 😉

**Comment by tmattio** [comment]
Thanks again 🎉

**Pull request merged**